Fixed crashing on exit in some cases

oleg68 opened the pull request:
In some situations, for example, if an unhandled exception occured (#1249), GrandOrgue exits and the exit causes a segfault.

The reason is `GOApp::OnExit` is called from wx when some windows, included `GOFrame` have not yet destroyed and continue to use objects `GOSound` and `GOConfig` that are deleted.

This PR forces destroying all windows before deleting the `GOSound` and `GOConfig`.

larspalo commented:
I've not had GO crash on exit recently (on Linux of course...) but with this PR I'm getting a warning at exit similar to:

`(GrandOrgue:3500): GLib-CRITICAL **: 21:25:34.004: Source ID 6590 was not found when attempting to remove it`

Looking at the code I'm wondering if the wxApp::CleanUp() shouldn't be done after wxApp::OnExit()? I understand that the call is only made to get the return value, but perhaps the order matter anyway...

oleg68 commented:
> I've not had GO crash on exit recently (on Linux of course...)

I reproduce this problem with adding ``throw wxT("Test exception");`` to the end of `GOOrganController::Load` before the `return`. Of cause, this exception is not a desired behavior, but even such exception occurs somewhere, GrandOrgue should not crash.



> but with this PR I'm getting a warning at exit similar to:
> 
> `(GrandOrgue:3500): GLib-CRITICAL **: 21:25:34.004: Source ID 6590 was not found when attempting to remove it`
> 
> Looking at the code I'm wondering if the wxApp::CleanUp() shouldn't be done after wxApp::OnExit()?

Yes, you are right. Normally `wxApp::CleanUp()`is called after `wxApp::OnExit()`, and with this PR `wxApp::CleanUp()` starts being called twice that causes the gtk warning.

oleg68 commented:
@larspalo I've reworked this PR. Instead of calling wxApp::Cleanup() from GOApp::OnExit(), I overrided wxApp::Cleanup() with GOApp:Cleanup() and moved the deletion code here from GOApp::OnExit() after calling wxApp::Cleanup()
